website/citations/__init__.py

```diff
@@ -0,0 +1,62 @@
+import os
+
+from citeproc import CitationStylesStyle, CitationStylesBibliography
+from citeproc import Citation, CitationItem
+from citeproc import formatter
+from citeproc.source.json import CiteProcJSON
+
+from website.settings import CITATION_STYLES_PATH
+
+
+def render(node, style='apa'):
+    """Given a node, return a citation"""
+    data = _node_to_csl_json(node)
+
+    bib_source = CiteProcJSON(data)
+
+    bib_style = CitationStylesStyle(os.path.join(CITATION_STYLES_PATH, style), validate=False)
+
+    bibliography = CitationStylesBibliography(bib_style, bib_source, formatter.plain)
+
+    citation = Citation([CitationItem(node._id)])
+
+    bibliography.register(citation)
+
+    def warn(citation_item):
+        pass
+
+    bibliography.cite(citation, warn)
+    return unicode(bibliography.bibliography()[0])
+
+
+def _node_to_csl_json(node):
+    """Given a node, return a dict in CSL-JSON schema
+
+    For details on this schema, see:
+        https://github.com/citation-style-language/schema#csl-json-schema
+    """
+    return [{
+        'author': [
+            _user_to_csl_json(user) for user in node.visible_contributors
+        ],
+        'id': str(node._id),
+        'issued': _datetime_to_csl_json(node.date_modified),
+        'publisher': 'Open Science Framework',
+        'title': unicode(node.title),
```

Comment:
Node titles are only ensured to be unicode if they're loaded from the database. If they're set in the same process, there's no guarantee they'll be set to a unicode value.